### DISCUSSION TITLE
Saving metadata in large files take a lot of time. It's normal?

### DISCUSSION THREAD

**Discussion by artofnois** [Zeugma440/atldotnet#189] [Q&A] [answered]
If I'm not wrong, when we do a Track.Save, the program seek to an offset at the end of file and begin to write tags.
Why a big files ( a lot of mines are sized >1G to 12G  ) take a lot of time to save? 
Some times, save action take over 10minutes one file..... (with no progress option)
Doing the same action on Soundminer take over 1 o 2 minutes "only".

And other question. If the save fail, ...  what about the audio file? Form me the most important thing is the audio file.
The labels I can write again. Some audios I never will  repeat again ...

**Comment by Zeugma440** [comment, marked as answer]
First things first, you may be interested in 
- The [Performance section of the FAQ](https://github.com/Zeugma440/atldotnet/wiki/4.-FAQ-(Frequently-Asked-Questions)#FAQM.5)
- The dedicated page for [progress feedback](https://github.com/Zeugma440/atldotnet/wiki/Progress-feedback)

> If I'm not wrong, when we do a Track.Save, the program seek to an offset at the end of file and begin to write tags.

Well, that's highly dependent on how the file is structured. Sometimes it's just as you say, sometimes parts to edit are located near the beginning of the file, and editing them requires to move the whole chunk of audio data.

> And other question. If the save fail, ... what about the audio file? Form me the most important thing is the audio file.

That's up to your app to put in place a safe mechanism, for instance by editing a temporary file instead of the original one. See our discussion in #113

**Comment by artofnois** [comment]
I will take a look to those links you suggest me. Thanks a lot.
Teh second part: 99% of my files are wave 96k, 24 o 32 bits depth.

**Comment by artofnois** [comment]
Now  I can begin to try the save part....
